source/d3d8to9_device.cpp

```diff
@@ -971,15 +971,72 @@ HRESULT STDMETHODCALLTYPE Direct3DDevice8::ValidateDevice(DWORD *pNumPasses)
 }
 HRESULT STDMETHODCALLTYPE Direct3DDevice8::GetInfo(DWORD DevInfoID, void *pDevInfoStruct, DWORD DevInfoStructSize)
 {
-	UNREFERENCED_PARAMETER(DevInfoID);
-	UNREFERENCED_PARAMETER(pDevInfoStruct);
-	UNREFERENCED_PARAMETER(DevInfoStructSize);
-
 #ifndef D3D8TO9NOLOG
 	LOG << "Redirecting '" << "IDirect3DDevice8::GetInfo" << "(" << this << ", " << DevInfoID << ", " << pDevInfoStruct << ", " << DevInfoStructSize << ")' ..." << std::endl;
 #endif
 
-	return S_FALSE;
+	if (pDevInfoStruct == nullptr || DevInfoStructSize == 0)
+		return D3DERR_INVALIDCALL;
+
+	HRESULT hr;
+	IDirect3DQuery9 *pQuery;
```

Comment:
This should be initialized to nullptr since you check below if it is null.

Reply:
Done, sorry for the oversight. I'm used to dealing with some form of smart pointers usually, so this slipped my mind.

Reply:
NP.  Thanks.